packages/common/src/failure.ts

```diff
@@ -357,7 +357,8 @@ export function ensureTemporalFailure(err: unknown): TemporalFailure {
   if (err instanceof TemporalFailure) {
     return err;
   } else if (err instanceof Error) {
-    const failure = new ApplicationFailure(err.message, err.name, false);
+    const name = err.constructor?.name || err.name;
```

Comment:
Should we even use `err.name` anymore? Probably makes sense not to.

Reply:
@bergundy I think it can't hurt to have it as a fallback. You can overwrite `constructor`:

```
const err = new Error('test');
err.constructor = null;
err.constructor?.name; // undefined
```

Reply:
Yeah, but I wouldn't care about this too much TBH.
I'd probably change this to `err.construtor.name ?? 'Error'`

Reply:
You know what, let's default to `err.name` as you suggested, looks like it should inherit from `Error`.